app/dao/api_key_dao.py

```diff
@@ -0,0 +1,51 @@
+from flask import current_app
+from itsdangerous import URLSafeSerializer
+
+from app import db
+from app.models import ApiKey
+
+
+def save_model_api_key(api_key, update_dict={}):
+    if update_dict:
+        del update_dict['id']
```

Comment:
Adam highlighted a problem with this line, if the 'id' is not in the dict object, this throws a KeyError exception.
Can you check for the key before deleting.